### PR TITLE
Fix filter dump in Data Modeling

### PR DESCRIPTION
**Pull request by doctrino**
### Fixed
- Bug when `cognite.client.data_classes.filter` used with any `data_modeling` endpoint raised a `CogniteAPIError` for 
  snake_cased properties. This is now fixed. 

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/filters.py [hunk 1]
```diff
@@ -68,8 +68,20 @@ def _dump_property(property_: PropertyReference, camel_case: bool) -> list[str]
 class Filter(ABC):
     _filter_name: str
 
-    def dump(self, camel_case: bool = True) -> dict[str, Any]:
-        return {self._filter_name: self._filter_body(camel_case)}
+    def dump(self, camel_case_property: bool = False) -> dict[str, Any]:
```
Comment (doctrino) [inline cognite/client/data_classes/filters.py:71]: This is the bug introduced in v7, the default was changed from `False` to `True`

File: cognite/client/data_classes/data_modeling/ids.py [hunk 2]
```diff
@@ -193,7 +194,7 @@ def create_args(id_: Id) -> tuple[str, str, str | None, Literal["node", "edge"]
                 return id_[0], id_[1], None, id_type  # type: ignore[return-value]
             raise ValueError("Instance given as a tuple must have two elements (space, externalId)")
         if isinstance(id_, tuple):
-            return id_[0], id_[1], id_[2] if len(id_) == 3 else None, None
+            return id_[0], id_[1], (id_[2] if len(id_) == 3 else None), None
```
Comment (haakonvt) [inline cognite/client/data_classes/data_modeling/ids.py:197]: Not sure if this is more readable or not, you choose 😜 
```suggestion
            return (*id_, None, None)[:4]
```

File: cognite/client/data_classes/relationships.py [hunk 3]
```diff
@@ -95,9 +95,9 @@ def _validate_resource_type(self, resource_type: str | None) -> None:
     def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> Relationship:
         instance = super()._load(resource, cognite_client)
         if instance.source is not None:
-            instance.source = instance._convert_resource(instance.source, instance.source_type)  # type: ignore
+            instance.source = instance._convert_resource(instance.source, instance.source_type, cognite_client)  # type: ignore
         if instance.target is not None:
-            instance.target = instance._convert_resource(instance.target, instance.target_type)  # type: ignore
+            instance.target = instance._convert_resource(instance.target, instance.target_type, cognite_client)  # type: ignore
```
Comment (haakonvt) [inline cognite/client/data_classes/relationships.py:100]: Nice catch!

File: cognite/client/_api/data_modeling/spaces.py [hunk 4]
```diff
@@ -63,18 +64,18 @@ def __iter__(self) -> Iterator[Space]:
         return self()
 
     @overload
-    def retrieve(self, spaces: str) -> Space | None:  # type: ignore[overload-overlap]
+    def retrieve(self, spaces: str) -> Space | None:
```
Comment (haakonvt) [inline cognite/client/_api/data_modeling/spaces.py:67]: Oooh nice! ...to get overloads working 👌

File: cognite/client/data_classes/filters.py [hunk 5]
```diff
@@ -68,8 +68,20 @@ def _dump_property(property_: PropertyReference, camel_case: bool) -> list[str]
...
+        """
+        Dump the filter to a dictionary.
+
+        Args:
+            camel_case_property (bool): Whether to camel case the property names. Defaults to False. Typically,
+                when the filter is used in data modeling, the property names should not be changed,
+                while when used with Assets, Event, Sequences, or Files, the property names should be camel cased.
```
Comment (haakonvt) [inline cognite/client/data_classes/filters.py:78]:
```suggestion
                while when used with Assets, Events, Sequences, or Files, the property names should be camel cased.
```

**Review by haakonvt — APPROVED** [review]
I like the change to `camel_case_property` to make it distinct from other methods

File: cognite/client/data_classes/data_modeling/ids.py (hunk 2 above)
Comment (erlendvollset) [reply cognite/client/data_classes/data_modeling/ids.py:197]: Can we catch this in a test?

Comment (erlendvollset) [reply cognite/client/data_classes/data_modeling/ids.py:197]: (nvm i see it's already done 💯)

Comment (haakonvt) [reply cognite/client/data_classes/data_modeling/ids.py:197]: @erlendvollset That change was only for easier readability

**Pull request merged**